onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -248,6 +248,12 @@ std::unique_lock<OrtMutex> TensorrtExecutionProvider::GetApiLock() const {
   return std::unique_lock<OrtMutex>(singleton);
 }
 
+#ifdef ORT_RUN_EXTERNAL_ONNX_TESTS
+// instantiate global unused builder object which keeps the TRT kernel library in memory
+// so that subsequent builders avoid the expensive load / unload process.
+auto const placeholder = tensorrt_ptr::unique_pointer<nvinfer1::IBuilder>(nvinfer1::createInferBuilder(GetTensorrtLogger()));
```

Comment:
if it's only relevant for tests, would it be possible to put this global in a test source file? can the variable name be more specific? and perhaps make it static/in the unnamed namespace?

Reply:
no, we only want TensorRT EP to depend on TensorRT libs, and not the OnnxRuntime test executable.

Reply:
can make the variable ort-trt specific. and i can put it in an anonymous namespace or make it static.